float32 center_x
float32 radius
float32 center_z
